### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by Alex-Vasilev**
-Fixed chats
-2FA implementation

### CODE REVIEW

File: models/chat.js [hunk 1]
```diff
@@ -5,14 +5,20 @@ const Schema = mongoose.Schema;
...
 const chatSchema = new Schema(
     {
+        title: String,
         messages: [{
             type: Schema.Types.ObjectId,
             ref: CONFIG.DB_MODELS.MESSAGE
+        }],
+        users: [{
+            type: Schema.Types.ObjectId,
+            ref: CONFIG.DB_MODELS.USER
```
Comment (esinitsa) [inline models/chat.js]: AMAZING! ⚡️

File: models/user.js [hunk 2]
```diff
@@ -20,7 +20,9 @@ const userSchema = new Schema(
         chats: [{
             type: Schema.Types.ObjectId,
             ref: CONFIG.DB_MODELS.CHAT
-        }]
+        }],
+        authyId: Number,
```
Comment (esinitsa) [inline models/user.js]: hhhmmmm, what is it?

File: routes/login.js [hunk 3]
```diff
@@ -1,77 +1,147 @@
...
-      User
-        .findOne({ name: req.body.name })
-        .populate({
-          path: 'chats',
-          populate: {
-            path: 'messages',
-            populate: 'user'
+  const { name } = req.body;
+
+  User
```
Comment (esinitsa) [inline routes/login.js]: move DB functionality to different files for all routes

File: routes/login.js [hunk 4]
```diff
@@ -1,77 +1,147 @@
...
-router.route('/registration').post((req, res, next) => {
-  const newUser = new User({
-    name: req.body.name,
-    password: req.body.password
+router.route('/registration').post((req, res) => {
+  const { name, password, email } = req.body;
+
+  User.findOne({ name }).exec((err, user) => {
+    if (err) {
+      res.json(err);
```
Comment (esinitsa) [inline routes/login.js]: I think that it is a bad way to send fully error to clients.

File: routes/login.js [hunk 5]
```diff
@@ -1,77 +1,147 @@
...
+        email,
+        authyId: regRes.user.id
+      });
+
+      newUser.save((err, usr) => {
+        if (err) {
+          res.json(err);
+        } else {
+          res.json({
+            ...JSON.parse(JSON.stringify(usr)),
```
Comment (esinitsa) [inline routes/login.js]: Good way, but what is the reason?

File: routes/login.js [hunk 6]
```diff
@@ -1,77 +1,147 @@
...
+      }
+
+      const newUser = new User({
+        name,
+        password,
+        email,
+        authyId: regRes.user.id
+      });
+
+      newUser.save((err, usr) => {
```
Comment (esinitsa) [inline routes/login.js]: bad naming, we can use `response` word for that, but it also too bad, but not `usr`

File: routes/login.js (hunk 5 above)
Comment (JeniaSavenko) [inline routes/login.js]: I don't understand why this
{...usr} not the same?
maybe I'm wrong.

File: models/user.js (hunk 2 above)
Comment (Alex-Vasilev) [reply models/user.js]: id for authy) 2fa

**Pull request merged**